static/app/views/insights/common/utils/useStarredTransactions.tsx

```diff
@@ -0,0 +1,65 @@
+import {
+  addErrorMessage,
+  addLoadingMessage,
+  addSuccessMessage,
+} from 'sentry/actionCreators/indicator';
+import {t} from 'sentry/locale';
+import {useMutation} from 'sentry/utils/queryClient';
+import useApi from 'sentry/utils/useApi';
+import useOrganization from 'sentry/utils/useOrganization';
+
+export type StarTransactionParams = {
+  projectId: string;
+  segmentName: string;
+};
+
+const URL_PREFIX = '/insights/starred-segments/';
+
+export function useStarredTransactions() {
+  const organization = useOrganization();
+  const api = useApi();
+  const url = `/organizations/${organization.slug}${URL_PREFIX}`;
+
+  const {mutate: mutateStarTransaction, ...starTransactionResult} = useMutation({
+    mutationFn: (params: StarTransactionParams) =>
+      api.requestPromise(url, {
+        method: 'POST',
+        data: {
+          project_id: params.projectId,
+          segment_name: params.segmentName,
+        },
+      }),
+    onSuccess: () => addSuccessMessage(t('Transaction starred')),
+    onError: () => addErrorMessage(t('Failed to star transaction')),
```

Comment:
Is the star component state updated optimistically? If not, I think it should, and maybe there needs to be a callback here to "roll back" the update if the request fails